src/llmq/utils.h

```diff
@@ -46,38 +47,38 @@ namespace utils
 {
 
 // includes members which failed DKG
-std::vector<CDeterministicMNCPtr> GetAllQuorumMembers(Consensus::LLMQType llmqType, const CBlockIndex* pQuorumBaseBlockIndex, bool reset_cache = false);
+std::vector<CDeterministicMNCPtr> GetAllQuorumMembers(Consensus::LLMQType llmqType, gsl::not_null<const CBlockIndex*> pQuorumBaseBlockIndex, bool reset_cache = false);
```

Comment:
so far as this and many other places requires not-null anyway, why even don't change `const CBlockIndex*> to the `const CBlockIndex&` ?

For reference, couple PR with bitcoin's similar changes of `CBlockIndex*` to `CBlockIndex&`. Bitcoin doesn't change it everywhere, but only when it is obviously non-nullptr, such as:
https://github.com/bitcoin/bitcoin/pull/25677/files
https://github.com/bitcoin/bitcoin/pull/25016/files

Reply:
Well; so one benefit imo is that dereferencing a nullptr is undefined behavior not guaranteed to be caught at runtime; additionally, gsl::not_null doesn't require additional syntax to pass to a later func that needs a ptr or additional syntax when passing in a ptr. In bitcoin PR you mention it appears they force to use something like `*Assert(m_chain.Tip()` which is significantly more typing :D

Reply:
well, `*Assert(...)` is 7 bytes shorter than `gsl::not_null<...>(...)`

And indeed, you can avoid UB by validating it before converting to ference.

Reply:
yeah, but once you convert pointer to reference, you can assume that all recursive calls; submethods; helpers, etc are never have nullptr anymore; but for case of pointer - you need to add gsl:not_null to every method do show that is not nullptr.

Reply:
Well yes; same as every method needs to be declared as a reference. Once a not_null is constructed it is zero cost to pass it to another not_null function